### PR TITLE
[BP-1.13][FLINK-22001] Fix forwarding of JobMaster exceptions to user

### PR DESCRIPTION
**Pull request by tillrohrmann**
Backport of #15715 to `release-1.13`.

### CODE REVIEW

File: flink-tests/src/test/resources/log4j2-test.properties [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 # Set root logger level to OFF to not flood build logs
 # set manually to INFO for debugging purposes
-rootLogger.level = OFF
+rootLogger.level = INFO
```
Comment (rmetzger) [inline flink-tests/src/test/resources/log4j2-test.properties:21]: We shouldn't commit this

**Review by rmetzger — APPROVED** [review]
Thanks a lot for the night shift! CI is passing, the change is fixing the problems described.

**Comment by tillrohrmann** [comment]
Thanks for the review @rmetzger. I will address your comment while merging this PR. The failing test case

**Comment by tillrohrmann** [comment]
Manually merged via ca9595317ba984120da00bc284c6109d0b4655d4

**Pull request closed without merging**